emcc.py

```diff
@@ -1409,7 +1409,7 @@ def phase_linker_setup(options, state, newargs, settings_map):
     # of LTO.
     if final_suffix in EXECUTABLE_ENDINGS:
       diagnostics.warning('emcc', '-shared/-r used with executable output suffix. This behaviour is deprecated.  Please remove -shared/-r to build an executable or avoid the executable suffix (%s) when building object files.' % final_suffix)
-    else:
```

Comment:
This change LGTM.. can we land this without the `cmake/Modules/Platform/Emscripten.cmake ` change though.  That looks separate and perhaps less obviously better.

Can you add a test for this too?  Maybe looks for the existing test for warning in question (`'linking a library with `-shared` will emit a static object file`) and modify that test?

Reply:
Ok, I'll move changes in `cmake/Modules/Platform/Emscripten.cmake` into another PR.